android/src/main/java/com/tanguyantoine/react/MusicControlModule.java

```diff
@@ -1,194 +1,306 @@
-        intent.setAction(intentAction);
-        PendingIntent pendingIntent = PendingIntent.getService(reactContext, 1, intent, 0);
-        return new Notification.Action.Builder(icon, title, pendingIntent).build();
+    @ReactMethod
+    public void setPlayback(ReadableMap info) {
+        if(!init) init();
+
+        long elapsedTime = info.hasKey("elapsedTime") ? (long)(info.getDouble("elapsedTime") * 1000) : 0;
+        long bufferedTime = info.hasKey("bufferedTime") ? (long)(info.getDouble("bufferedTime") * 1000) : 0;
+        float speed = info.hasKey("speed") ? (float)info.getDouble("speed") : 1;
+        int state = info.hasKey("state") ? info.getInt("state") : PlaybackStateCompat.STATE_NONE;
+        int vol = info.hasKey("volume") ? info.getInt("volume") : 100;
+
+        pb.setState(state, elapsedTime, speed);
+        pb.setBufferedPosition(bufferedTime);
+
+        isPlaying = state == PlaybackStateCompat.STATE_PLAYING || state == PlaybackStateCompat.STATE_BUFFERING;
+        notification.show(nb, isPlaying);
+
+        PlaybackStateCompat playbackState = pb.build();
+        session.setPlaybackState(playbackState);
+        session.setPlaybackToRemote(volume.create(null, vol));
     }
 
-    private void updateNotification(){
-        initNotificationBuilder();
+    @ReactMethod
+    public void resetNowPlaying() {
+        if(!init) return;
+        if(artworkThread != null && artworkThread.isAlive()) artworkThread.interrupt();
+        artworkThread = null;
 
-        this.notificationBuilder
-                .setSmallIcon(android.R.drawable.arrow_up_float);
-        String title = infos.hasKey("title") ? infos.getString("title") : "";
-        String content = infos.hasKey("artist") ? infos.getString("artist") : "";
-        this.notificationBuilder
-                .setContentTitle(title)
-                .setContentText(content);
+        md = new MediaMetadataCompat.Builder();
+        pb = new PlaybackStateCompat.Builder();
 
-        if(shouldActivateControl(CONTROL_previousTrack)){
-            this.notificationBuilder
-                    .addAction(generateAction(android.R.drawable.ic_media_previous, "Previous", ACTION_PREVIOUS));
-        }
-        //builder.addAction(generateAction(android.R.drawable.ic_media_rew, "Rewind", ACTION_REWIND));
-        if(shouldActivateControl(CONTROL_play)) {
-            this.notificationBuilder
-                    .addAction(generateAction(android.R.drawable.ic_media_play, "Play", ACTION_PLAY));
-        }
-        if(shouldActivateControl(CONTROL_pause)) {
-            this.notificationBuilder
-                    .addAction(generateAction(android.R.drawable.ic_media_pause, "Pause", ACTION_PAUSE));
+        notification.hide();
+        session.setActive(false);
+    }
+
+    @ReactMethod
+    public void enableControl(String control, boolean enable) {
+        if(!init) init();
+
+        long controlValue;
+        switch(control) {
+            case "nextTrack":
```

Comment:
The deleted static fields were from the old android support. @cinder92, you should fix your workspace to target the current head instead of merging the new commits.

The only thing he added is from line 291 to 305 and lines 282, 283. The rest is just merging problems that he should fix